### PR TITLE
Rename and structure unit test and perf tests

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: #1696 

Rename the performance fw to only `Perf` 
Structure for unit tests under test/ut folder and test/per for performance tests
Makes each service to include its test projects instead of including the test projects from cmake root file
Fixes warnings from unit tests (visible until now because the tests were added before setting the global compiler options)

### CODE REVIEW

**Comment by vhvb1989** [comment]
azp run cpp - keyvault

File: sdk/storage/azure-storage-blobs/test/perf/src/main.cpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#include <azure/perf.hpp>
+
+#include "azure/storage/blobs/test/perf/download_blob.hpp"
+
+int main(int argc, char** argv)
+{
+
+  // Create the test list
+  std::vector<Azure::Perf::TestMetadata> tests{
+      Azure::Storage::Blobs::Test::Performance::DownloadBlob::GetTestMetadata()};
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/test/perf/src/main.cpp:13]: There's no `Test` namespace under `Blobs`. We only have one `Test` namespace, so `Azure::Storage::Test::Performance`

**Review by Jinming-Hu — APPROVED** [review]
changes to storage lgtm

File: sdk/storage/azure-storage-blobs/test/perf/src/main.cpp (hunk 1 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/test/perf/src/main.cpp:13]: On a second thought, I don't think `Performance` is necessary at all. We don't have `Azure::Storage::Test::Ut`, right?

perf test and ut test are two separate executables. We don't mix the source files, so there's no need to distinguish them with different namespaces.

**Comment by Jinming-Hu** [comment]
@vhvb1989 Hi, when can I expect this PR to be merged? I'm starting working on perf test and stress test.

**Comment by vhvb1989** [comment]
> @vhvb1989 Hi, when can I expect this PR to be merged? I'm starting working on perf test and stress test.

Just waiting from some Azure Core approval.
I will commit your suggestion for having only one spacename `Test` and fix the current conflicts

**Comment by vhvb1989** [comment]
Enabling auto-merge to unblock Jinming. I can address Azure Core concerns (if any) as soon as they review.
Also making the call since this is only test code

**Pull request merged**